src/azure-cli/azure/cli/command_modules/batch/_client_factory.py

```diff
@@ -51,8 +51,7 @@ def batch_data_client_factory(cli_ctx, kwargs):
     if not token_credential and not account_key:
         from azure.cli.core._profile import Profile
         profile = Profile(cli_ctx=cli_ctx)
-        resource = cli_ctx.cloud.endpoints.batch_resource_id
-        token_credential, _, _ = profile.get_login_credentials(resource=resource)
+        token_credential, _, _ = profile.get_login_credentials()
```

Comment:
Please double check if `cli_ctx.cloud.endpoints.batch_resource_id` needs to be set as `credential_scopes` when creating `BatchClient`.